src/cpu.rs

```diff
@@ -75,8 +131,6 @@ impl XRegisters {
     /// Create a new `XRegisters` object.
     pub fn new() -> Self {
         let mut xregs = [0; REGISTERS_COUNT];
-        // The stack pointer is set in the default maximum memory size + the start address of dram.
-        xregs[2] = DRAM_BASE + DRAM_SIZE;
```

Comment:
Since the stack is based on where the system wants it, we don't know where we would have to initialize this, and it needs to be done elsewhere.  When the DRAM is initialized on the bus, I do set the stack pointer to the end of ram.